### DISCUSSION TITLE
[RFE] make bcachefs ready

### DISCUSSION THREAD

**Discussion by hgkamath** [stratis-storage/project#708] [Ideas]
As of today bcachefs is yet to be mainlined.
I am guessing that it will eventually be in your radar. I did not see a bug or discussion regarding this, of whether it is in the roadmap or works.
bcachefs was still very new when stratis was first rolled out and much has happened since then.
I think this entails, allowing for replacement plumbing of api calls/tool-cmd invocations from XFS handling to corresponding bcachefs handling, so that bcachefs can be experimented with as a XFS alternative. 
Filing this bug, so that one can be notified when the day comes that stratisd supports bcachefs in addition to XFS, if that ever happens. Maybe draw up a checklist of what bcachefs should be able to do before it can be considered.

**Comment by NickCao** [comment]
Wouldn't bcachefs natively support multiple devices, caching, encryption and snapshots, in other words, it's a direct competitor of this project, rather than a possible enhancement.

**Comment by hgkamath** [comment]
I hadn't thought of it that way. In opensource, things don't have to be competitors, just alternate ways of doing things, each with their own pros and cons. For whatever reasons someone may prefer stratis's volume management over bcachefs's native volume management, for its features, and may prefer bcachefs over XFS because bcachefs can use an SSD partition as a write caching device for performance.
This was just a spontaneous curiosity question for me. I may not know much to add anything to discussion.

**Comment by NickCao** [comment]
Sorry for my words for being too strong and I did not mean that, and I would also love to see these two projects work together.
